server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -575,36 +614,60 @@ public void dangerPermanentlyRemoveSecretsDeletedBeforeDate(DateTime deletedBefo
     }
   }
 
+  private Long getOwnerId(Configuration configuration, String owner) {
+    GroupDAO groupDAO = groupDAOFactory.using(configuration);
+    Optional<Group> group = groupDAO.getGroup(owner);
+    return group.isPresent() ? group.get().getId() : null;
+  }
```

Comment:
Oops, I think this is a bug: if the caller specifies an owner which doesn't exist, we should fail rather than silently succeed.